channel.c

```diff
@@ -226,13 +226,16 @@ static char * get_scan_element(const struct iio_channel *chn, size_t *length)
 /* Returns a string containing the XML representation of this channel */
 char * iio_channel_get_xml(const struct iio_channel *chn, size_t *length)
 {
-	size_t len = sizeof("<channel id=\"\" name=\"\" "
-			"type=\"output\" ></channel>")
-		+ strlen(chn->id) + (chn->name ? strlen(chn->name) : 0);
-	char *ptr, *str, **attrs, *scan_element = NULL;
+	ssize_t len;
+	char *ptr, *eptr, *str, **attrs, *scan_element = NULL;
 	size_t *attrs_len, scan_element_len = 0;
 	unsigned int i;
 
+	len = sizeof("<channel id=\"\" type=\"\" ></channel>");
```

Comment:
for important files like `channel.c` this patch is doing too much in one go;
i can understand the value of converting 
```
	size_t len = sizeof("<channel id=\"\" name=\"\" "
			"type=\"output\" ></channel>")
		+ strlen(chn->id) + (chn->name ? strlen(chn->name) : 0);
```

to
```
	len = sizeof("<channel id=\"\" type=\"\" ></channel>");
	len += strnlen(chn->id, MAX_CHN_ID);
	len += chn->is_output ? sizeof("output") : sizeof("input");
	len += chn->name ? sizeof(" name= ") + strnlen(chn->name, MAX_CHN_NAME) : 0;
```

but it should be in it's own patch;
otherwise we risk slipping stuff at review;

Reply:
sure - I can split up this commit into multiple.